code/game/objects/items/devices/mat_synth.dm

```diff
@@ -61,37 +62,43 @@
+									var/transfer_amount = min(created_sheet.max_amount - created_sheet.amount, amount)
+									created_sheet.amount += (transfer_amount-1)
+									amount -= transfer_amount
+								if(amount >= 1 && (created_sheet.amount >= created_sheet.max_amount))
+									to_chat(R, "<span class='warning'>Dropping [amount], you cannot hold anymore of [initial(material_type.name)].</span>")
+									var/obj/item/stack/sheet/dropped_sheet = new material_type(get_turf(src))
+									dropped_sheet.amount = (amount - 1)
 						else
-							if(created_sheet.amount <= created_sheet.max_amount)
-								var/transfer_amount = min(created_sheet.max_amount - created_sheet.amount, amount)
-								created_sheet.amount += (transfer_amount-1)
-								amount -= transfer_amount
-							if(amount >= 1 && (created_sheet.amount >= created_sheet.max_amount))
-								to_chat(R, "<span class='warning'>Dropping [amount], you cannot hold anymore of [initial(material_type.name)].</span>")
-								var/obj/item/stack/sheet/dropped_sheet = new material_type(get_turf(src))
-								dropped_sheet.amount = (amount - 1)
-
+							if((inside_sheet.amount + amount) <= inside_sheet.max_amount)
+								inside_sheet.amount += amount
+								to_chat(R, "<span class='notice'>Added [amount] of [initial(material_type.name)] to the stack.</span>")
+								return
+							else
+								if(inside_sheet.amount <= inside_sheet.max_amount)
+									var/transfer_amount = min(inside_sheet.max_amount - inside_sheet.amount, amount)
+									inside_sheet.amount += transfer_amount
+									amount -= transfer_amount
+								if(amount >= 1 && (inside_sheet.amount >= inside_sheet.max_amount))
+									to_chat(R, "<span class='warning'>Dropping [amount], you cannot hold anymore of [initial(material_type.name)].</span>")
+									var/obj/item/stack/sheet/dropped_sheet = new material_type(get_turf(src))
+									dropped_sheet.amount = amount
 					else
-						if((inside_sheet.amount + amount) <= inside_sheet.max_amount)
-							inside_sheet.amount += amount
-							to_chat(R, "<span class='notice'>Added [amount] of [initial(material_type.name)] to the stack.</span>")
-							return
-						else
-							if(inside_sheet.amount <= inside_sheet.max_amount)
-								var/transfer_amount = min(inside_sheet.max_amount - inside_sheet.amount, amount)
-								inside_sheet.amount += transfer_amount
-								amount -= transfer_amount
-							if(amount >= 1 && (inside_sheet.amount >= inside_sheet.max_amount))
-								to_chat(R, "<span class='warning'>Dropping [amount], you cannot hold anymore of [initial(material_type.name)].</span>")
-								var/obj/item/stack/sheet/dropped_sheet = new material_type(get_turf(src))
-								dropped_sheet.amount = amount
+						to_chat(R, "<span class='notice'>Dropping [amount] of [initial(material_type.name)] onto the floor.</span>")
+						var/obj/item/stack/sheet/dropped_sheet = new material_type(get_turf(src))
+						dropped_sheet.amount = amount
```

Comment:
Also consider using `drop_stack` instead of `new [sheet]` like I do in #15243. And I've recently learned that you don't need `get_turf` for things like this.